src/Client.php

```diff
@@ -232,28 +260,36 @@ function () {
                     yield $channel->qos($this->config->qosSize, $this->config->qosCount, $this->config->qosGlobal);
 
                     asyncCall(function () use ($id) {
-                        /** @var Protocol\ChannelCloseFrame|Protocol\ChannelCloseOkFrame $frame */
-                        $frame = yield Promise\first([
-                            $this->await(Protocol\ChannelCloseFrame::class, $id),
-                            $this->await(Protocol\ChannelCloseOkFrame::class, $id)
-                        ]);
-
-                        $this->connection->cancel($id);
-
-                        if ($frame instanceof Protocol\ChannelCloseFrame) {
-                            $buffer = new Buffer;
-                            $buffer
-                                ->appendUint8(1)
-                                ->appendUint16($id)
-                                ->appendUint32(4)
-                                ->appendUint16(20)
-                                ->appendUint16(41)
-                                ->appendUint8(206);
-
-                            yield $this->connection->write($buffer);
+                        try {
+                            $frame = yield Promise\first([
+                                $this->await(Protocol\ChannelCloseFrame::class, $id),
+                                $this->await(Protocol\ChannelCloseOkFrame::class, $id)
+                            ]);
+
+                            $channel = $this->channels[$id];
+                            unset($this->channels[$id]);
+
+                            if ($frame instanceof Protocol\ChannelCloseFrame) {
+                                $buffer = new Buffer;
+                                $buffer
+                                    ->appendUint8(1)
+                                    ->appendUint16($id)
+                                    ->appendUint32(4)
+                                    ->appendUint16(20)
+                                    ->appendUint16(41)
+                                    ->appendUint8(206);
+
+                                yield $this->connection->write($buffer);
+                                $channel->forceClose(new ChannelException("Channel closed: {$frame->replyText}"));
+                            }
+                        }
+                        catch (MultiReasonException $exception) {
+                            // There was an error when waiting for those frames
```

Comment:
That was not handled before resulting in an exception that propagates to the loop but can't be handled in the application. Not very helpful